cxx/pclean/pclean.cc

```diff
@@ -8,12 +8,49 @@
 #include <random>
 
 #include "cxxopts.hpp"
+#include "irm.hh"
 #include "hirm.hh"
 #include "inference.hh"
+#include "util_io.hh"
+#include "pclean/csv.hh"
 #include "pclean/io.hh"
 #include "pclean/schema.hh"
 #include "pclean/schema_helper.hh"
 
+T_observations translate_observations(
+    const DataFrame& df, const T_schema &schema) {
+  T_observations obs;
+
+  for (const auto& col : df.data) {
+    const std::string& col_name = col.first;
```

Comment:
We generally haven't been using const on local variables  I don't have a strong opinion either way, but I think we should be consistent in using it everywhere/nowhere/some places according to some rules we agree on (and up to now we've defaulted to "nowhere"). WDYT? 

The Google style guide says "Using const on local variables is neither encouraged nor discouraged."

Reply:
I don't care much about this, but I do see some other uses of this in the code base, like on lines 267, 306 and 423 of util_io.cc.

And we use const all the time on for-loop variables, and those are local variables too!

I guess my overall opinion is that this might be an area where some inconsistency is fine.

Reply:
Inconsistency is fine by me too, I guess I'd just prefer that it feel less arbitrary. Are there any loose guidelines you'd propose?

Reply:
I guess my suggestion would be to lean towards using const when you can and when it would help.

So for example, if a local variable is only alive for a few lines and it is clear what is being done to it, the const doesn't help very much.  But for a medium sized or longer function, annotating something as const can give the reader some valuable info about what is going on.

I guess the other guideline I would suggest is that consistency at the function level is more important than any sort of global consistency.  That is:  if my function has three variables and I mark two of them as const, then there is a slight presumption that the third non-const variable gets mutated somewhere.  So don't do that unless it is.